app/uk/gov/hmrc/helptosavefrontend/config/CustomWSConfigParser.scala

```diff
@@ -84,31 +75,48 @@ class CustomWSConfigParser @Inject()(configuration: Configuration, env: Environm
         )
       )
     )
-    modded
-  }
 
-  private def mergeAllStores(config: Configuration): Configuration = {
-    mergeStores(mergeStores(config, "key"), "trust")
+    wsClientConfig
   }
 
-  private def mergeStores(config: Configuration, name: String): Configuration = {
-    val under = config.underlying
-    if (under.hasPath(s"play.ws.ssl.${name}Manager.store")) {
-      val singleStore: ConfigObject = under.getObject(s"play.ws.ssl.${name}Manager.store")
-      val stores: util.List[AnyRef] = under.getList(s"play.ws.ssl.${name}Manager.stores").unwrapped()
-      if (singleStore != null) {
-        stores.add(singleStore)
-      }
-      config.copy(underlying = config.underlying.withValue(s"play.ws.ssl.${name}Manager.stores", ConfigValueFactory.fromIterable(stores)))
-    } else config
+  private def createTrustStoreConfig(ts: TrustStoreConfig, data: String): TrustStoreConfig = {
+
+    createTempFileForData(data) match {
+      case Success(trustFile) ⇒
+        val keyStore = KeyStore.getInstance(KeyStore.getDefaultType)
+        keyStore.load(null, null)
+
+        val certs = generateCertificates(trustFile)
```

Comment:
could we not make

 `def createTempFileForData(data: String): Try[File]`

not have the type signature

` def createTempFileForData(data: String): Array[Byte]`

instead?  Then `generateCertificates` could be implemented like this:
```scala
private def generateCertificates(file: Array[Byte]): Seq[Certificate] = {
  val bais = new ByteArrayInputStream(file)
  try{
    CertificateFactory.getInstance("X.509").generateCertificates(bais).toList
  } finally {
  case NonFatal(_) =>
    bais.close()
  // handle error...
  }
}
```
Then we avoid having to read from file again which is an expensive operation